packages/client/wallets/aa/src/blockchain/wallets/smartAccount.ts

```diff
@@ -0,0 +1,49 @@
+import { logInfo } from "@/services/logging";
+import { gelatoBundlerProperties, usesGelatoBundler } from "@/utils";
+import { SmartAccountClient } from "permissionless";
+import { EntryPoint } from "permissionless/types/entrypoint";
+
+import { EVMBlockchainIncludingTestnet } from "@crossmint/common-sdk-base";
+
+export function toCrossmintSmartAccountClient<Client extends SmartAccountClient<EntryPoint>>({
+    crossmintChain,
+    smartAccountClient,
+}: {
+    crossmintChain: EVMBlockchainIncludingTestnet;
+    smartAccountClient: Client;
+}): Client {
+    return {
+        ...smartAccountClient,
+        async signMessage(args) {
+            try {
+                return await smartAccountClient.signMessage(args);
+            } catch (error) {
+                throw new Error(`Error signing message. If this error persists, please contact support.`);
+            }
+        },
+        async signTypedData(data) {
+            try {
+                return smartAccountClient.signTypedData(data);
+            } catch (error) {
+                throw new Error(`Error signing typed data. If this error persists, please contact support.`);
+            }
```

Comment:
I'm not sure we should be obfuscating the error like this, but copying behavior from existing functions for now. We can re-eval later.